client/src/components/Search.tsx

```diff
@@ -1,20 +1,22 @@
-import React, { useState } from "react";
+import React, { useState, useRef, useEffect } from "react";
+import { RoomSearchImpl } from "../logic/impl/RoomSearchImpl";
 
 function Search() {
-
   const [inputValue, setInputValue] = useState("");
   const [showDropdown, setShowDropdown] = useState(false);
-  const [dropdownOptions, setDropdownOptions] = useState([
-    "canteen",
-    "I5",
-    "I7",
-    "P1",
-    "P2",
-    "wc"
-  ]);
+  const [dropdownOptions, setDropdownOptions] = useState([]);
+
+  const searchRef = useRef(null);
+
+  const handleInputChange = async (event) => {
+    const inputValue = event.target.value;
+    setInputValue(inputValue);
+
+    const roomSearch = new RoomSearchImpl();
```

Comment:
A component should never create a specific implementation of the interface. It should just receive an interface as a prop (dependency injection).

That is a bit harder to do and I wasn't sure if it was possible so I tried and conformed it in #43.

In that pull request, you can see that only a root page (`Home` in your example) can create an implementation of an interface and all other components should receive it as a prop. That is more clearly described in the `React components` section in [the wiki](https://github.com/JakovTomasic/RitehMaps/wiki/Writing-code).

The advantages of this approach are:
- it makes later mocking and changing implementations really easy
- we can easily use different functions for each Search component. E.g. for start and destination searches.
- it follows the "Dependency Inversion" principle of the SOLID principles (if you are curious, I like [this](https://www.baeldung.com/solid-principles#d) explanation of the principle)